src/OrleansAzureUtils/MultiClusterNetwork/GossipTableInstanceManager.cs

```diff
@@ -0,0 +1,327 @@
+﻿using Microsoft.WindowsAzure.Storage.Table;
+using Orleans.AzureUtils;
+using Orleans.MultiCluster;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.MultiClusterNetwork
+{
+    /// <summary>
+    /// Low level representation details and functionality for Azure-Table-Based Gossip Channels
+    /// </summary>
+    internal class GossipTableEntry : TableEntity
+    {
+        // used for partitioning table
+        internal string GlobalServiceId { get { return PartitionKey; } }
+
+        public DateTime GossipTimestamp { get; set; }   // timestamp of gossip entry
+
+        #region gateway entry
+
+        public string Status { get; set; }
+
+        // all of the following are packed in rowkey
+
+        public string ClusterId;
+
+        public IPAddress Address;
+
+        public int Port;
+
+        public int Generation;
+
+        public SiloAddress SiloAddress;
+
+        #endregion
+
+        #region configuration entry
+
+        public string Clusters { get; set; }   // comma-separated list of clusters
+
+        public string Comment { get; set; }
+
+        #endregion
+
+        internal const string CONFIGURATION_ROW = "CONFIG"; // Row key for configuration row.
+
+        private const string Separator = ","; // safe because clusterid cannot contain commas
+        private readonly static char[] SeparatorChars = Separator.ToCharArray();
+        internal const string ClustersListSeparator = ","; // safe because clusterid cannot contain commas
+        private readonly static char[] ClustersListSeparatorChars = ClustersListSeparator.ToCharArray();
+        private const string RowKeyFormat = "{0}"+ Separator + "{1}" + Separator + "{2}" + Separator + "{3}";
+
+        public static string ConstructRowKey(SiloAddress silo, string clusterid)
+        {
+            return String.Format(RowKeyFormat, clusterid, silo.Endpoint.Address, silo.Endpoint.Port, silo.Generation);
+        }
+
+        internal void UnpackRowKey()
```

Comment:
`ParseSiloAddress` or `ParseSiloAddressFromRowKey` I think would be more explicit and help readability.